### PR TITLE
Why rasa skill

### LINKED ISSUE #508: Handle "why rasa"
Look for user inputs like 
- why should i use rasa
- why rasa
- why should i use rasa over others”
- how does rasa compare to dialogflow

etc. 

Some of the content could come from a page like [this](https://rasa.com/docs/rasa/migrate-from/ibm-watson-to-rasa/) (i.e. why rasa over cloud-based tools)

### PR DESCRIPTION
**Pull request by melindaloubser1**
Added "why rasa?" skill.
If a current_api is given it will also show the migration guide link.
Closes #508

### CODE REVIEW

File: actions/actions.py [hunk 1]
```diff
@@ -814,3 +814,24 @@ def run(self, dispatcher, tracker, domain) -> List[EventType]:
         tag_convo(tracker, label)
 
         return []
+
+
+class ActionWhyRasa(Action):
```
Comment (b-quachtran) [inline actions/actions.py:819]: `ActionWhyRasa` isn't being used anywhere in the stories. Can this custom action be removed?

File: actions/actions.py [hunk 2]
```diff
@@ -814,3 +814,24 @@ def run(self, dispatcher, tracker, domain) -> List[EventType]:
...
+
+        reasons = "\n" + "\n- ".join(
+            [
+                "Rasa is Open Source!",
+                "Rasa invests in extensive research to create leading-edge conversational AI",
+                "Rasa's NLU provides developers with the technology to understand messages, determine intent, and capture key contextual information.",
+                "Assistants built on Rasa can hold meaningful conversations with users - multi-step conversations that remember context and integrate business logic. ",
+            ]
+        )
+        dispatcher.utter_message(template="utter_why_rasa", reasons=reasons)
```
Comment (b-quachtran) [inline actions/actions.py:835]: `reasons` param is being passed in for the `utter_why_rasa` template but I don't see it in the domain.yml file

File: domain.yml [hunk 3]
```diff
@@ -543,6 +544,33 @@ responses:
...
   utter_no_further_info:
   - text: Sorry, that's all the info I've got on that subject 😕
+  utter_why_rasa:
+  - text:  Rasa provides the infrastructure and tools needed to build
+      high-performing, resilient assistants that work. What's more,
+      Rasa is **open source** - no black-box magic! 
+  utter_ask_more: 
+  - text: Would you like to learn more?
+  utter_ask_more_migrate:
+  - text: Would you like to learn more advantages of switching to Rasa?
```
Comment (b-quachtran) [inline domain.yml:554]:
```suggestion
  - text: Would you like to learn more about the advantages of switching to Rasa?
```

File: domain.yml [hunk 4]
```diff
@@ -543,6 +544,33 @@ responses:
...
+  utter_why_rasa_research:
+    - text: Rasa invests in extensive research to create leading-edge 
+        conversational AI, enabling developers to create the best text- and
+        voice-based assistants, without a dedicated research team.
```
Comment (b-quachtran) [inline domain.yml:558]:
```suggestion
        voice-based assistants without a dedicated research team.
```

File: actions/actions.py (hunk 2 above)
Comment (melindaloubser1) [reply actions/actions.py:835]: sorry this whole action is unnecessary! will remove

File: actions/actions.py (hunk 1 above)
Comment (melindaloubser1) [reply actions/actions.py:819]: yes

**Pull request merged**